### DISCUSSION TITLE
Wagmi 0.9.x with ts-node and Nextjs

### DISCUSSION THREAD

**Discussion by Toesmash** [wevm/wagmi#1480] [Q&A] [answered]
hi wagmi friends 👋 

i am running a NextJS application in Typescript with custom server.
since upgrading from `wagmi: 0.7.15` to `wagmi: 0.9.2` version i am having troubles running my project in development mode with ts-node.

i am getting runtime error from NextJs _app file (which contains basic wagmi init)

```
Error: require() of ES Module xxx/node_modules/wagmi/dist/index.js from xxx/build/server/pages/_app.js not supported.
Instead change the require of index.js in xxx/build/server/pages/_app.js to a dynamic import() which is available in all CommonJS modules.
```

here is the setup code i use

```json
tsconfig.json
{
  "compilerOptions": {
    "target": "esnext",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "incremental": true,
    "jsx": "preserve"
  },
  "ts-node": {
    "transpileOnly": true,
    "compilerOptions": {
      "module": "commonjs"
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules", "build", "out", ".next", "next.config.js"]
}

```

```json
tsconfig.server.json
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "commonjs",
    "outDir": "build/server",
    "target": "ES2016",
    "isolatedModules": false,
    "noEmit": false
  },
  "include": ["server/**/*", "./environment.d.ts"]
}

```

```json
package.json
"dev": "ts-node server/index.ts",
"build:server": "tsc --project tsconfig.server.json",
"build:next": "next build",
"build": "npm run build:next && npm run build:server",
```

i know that wagmi dropped commonjs support from 0.8.x version. the app can be built for production, but does throw the error in development environment (using ts-node). i cannot get the project running without ts-node compilerOptions module set to commonjs. so far it seems my only option is to downgrade, but would love to use the newer versions.

appreciate any help :)

**Comment by jxom** [comment, marked as answer]
Does the `--esm` flag work?

https://github.com/TypeStrong/ts-node#esm

**Reply by Toesmash** [reply]
unfortunately, i still get error with --esm flag

**Reply by Toesmash** [reply]
i managed to get it running, and the esm flag was also part of the solution 👍 

few things i changed:
- package.json file needs to have its module type defined - `"type": "module"`
- setup node-ts with esm enabled, without compilerOptions (in tsconfig.json)
```json
  "ts-node": {
    "esm": true,
    "transpileOnly": true,
    "experimentalResolver": true,
    "experimentalSpecifierResolution": "node"
  }
```
- exports in correct format - in my case there was a single line in next config file which was in commonjs
```diff
// next.config.js
- module.exports = nextConfig;
+ export default nextConfig;
```
